router/dataplane_spec.gobra

```diff
@@ -343,3 +346,369 @@ func closureSpec2(i uint16, c BatchConn)
 requires true
 func closureSpec3(c BatchConn)
 /** End of closure specs for the Run method **/
+
+/** Start of io-spec helper functions **/
+
+ghost 
+requires 0 < seg1Len
+requires 0 <= seg2Len
+requires 0 <= seg3Len
+requires currHF < seg1Len + seg2Len + seg3Len
+decreases
+pure func computeSeqLen(currHF int, seg1Len int, seg2Len int, seg3Len int) int{
+	return seg1Len > currHF ? seg1Len : ((seg1Len + seg2Len) > currHF ? seg2Len : seg3Len)
```

Comment:
`computeSeqLen` is a misleading name. I would call it sth like `lengthOfCurrSeg`. Also, I wonder if we need to attach all these preconditions. I know what you mean by them, but here they are only adding extra requirements that must be checked, even though this call can never fail regardless of the values of the outputs. The advantage is that we can get more localized error messages, if it is indeed the case that this condition must hold on all call sites